docroot/modules/humsci/hs_bugherd/src/Plugin/rest/resource/BugherdResource.php

```diff
@@ -193,9 +194,81 @@ protected function bugherdTaskUpdated(array $task) {
         $issue->addComment($this->t('Issue closed in Bugherd by @name', ['@name' => $task['updater']['display_name']]));
       }
     }
+
+    // Attach the screenshot to the Jira Issue if it doesn't have any already.
+    if (!empty($task['screenshot_url']) && empty($issue->fields->attachment)) {
+      $this->addAttachment($issue->getKey(), $task['screenshot_url']);
+    }
+
     return $issue->getKey();
   }
 
+  /**
+   * Add an attachment file to a JIRA issue.
+   *
+   * The method Drupal\jira_rest\JiraRestWrapperService::attachFileToIssueByKey
+   * doesn't work correctly. This provides our own method to attach the file.
+   * https://www.drupal.org/project/jira_rest/issues/2982894 is the issue.
+   *
+   * @param string $issueKey
+   *   Jira Issue ID.
+   * @param string $file
+   *   Url to file.
+   * @param null $name
+   *   Desired name of the file.
+   *
+   * @return mixed
+   *
+   * @see https://community.atlassian.com/t5/Jira-questions/Upload-URL-Attachment-via-API-on-JIRA-PHP/qaq-p/44828
+   * @see https://developer.atlassian.com/cloud/jira/platform/rest/?_ga=2.240914371.1127742332.1530222923-1785182880.1526662104#api-api-2-issue-issueIdOrKey-attachments-post
+   */
+  protected function addAttachment($issueKey, $file, $name = NULL) {
+    $jira_config = $this->configFactory->get('jira_rest.settings');
+    $url = $jira_config->get('jira_rest.instanceurl');
+    $username = $jira_config->get('jira_rest.username');
+    /** @var \Drupal\key\Entity\Key $key */
+    $key = Key::load($jira_config->get('jira_rest.password'));
+    $password = $key ? $key->getKeyValue() : '';
+
+    // Creating file name.
+    $file_name = $name ? $name : time() . '-' . basename($file);
+    $path = tempnam(sys_get_temp_dir(), 'bugherd');
+
+    // Saving file in a temp path.
+    file_put_contents($path, file_get_contents($file));
+
+    // Initiating CURLFile for preparing the upload
+    $cfile = new \CURLFile($path);
```

Comment:
🤔 This is a PHP class but do you still add a use statement? Not going to worry about it blocking this but it would be good to know what the convention is.

Reply:
hmm. that would be a good thing to find out.